### PR TITLE
1050: bmcweb: Add PCIeSlot to Associated assembly link (#343)

### PR DESCRIPTION
**Pull request by baemyung**
This commit adds PCIeSlot to NVMe backplane assembly link. The link is listed in Oem property for each of the configured PCIe slots.

This is downstream only.
Issue: https://github.com/ibm-openbmc/dev/issues/3535

Testing:
- Validator PASSED

- List properties for all PCIe Slots

```
$ curl -k -X GET https://${bmc}/redfish/v1/Chassis/chassis/PCIeSlots
{
  "@odata.id": "/redfish/v1/Chassis/chassis/PCIeSlots",
  "Slots": [
    {
      "HotPluggable": false,
      "Lanes": 0,
      "Links": {
        "Oem": [
          {
            "@odata.id":
"/redfish/v1/Chassis/chassis/Assembly#/Assemblies/24"
          }
        ],
        "PCIeDevice": [
          {
            "@odata.id": "/redfish/v1/Systems/system/PCIeDevices/drive0"
          }
        ]
      },
    },
...
}

$ curl -k -X GET https://${bmc}/redfish/v1/Chassis/chassis/Assembly#/Assemblies/24
{
...
    {
      "@odata.id":
"/redfish/v1/Chassis/chassis/Assembly#/Assemblies/24",
      "@odata.type": "#Assembly.v1_3_0.AssemblyData",
      "Location": {
        "PartLocation": {
          "ServiceLabel": "U78DB.ND0.WZS002U-P1"
        }
      },
      "LocationIndicatorActive": false,
      "MemberId": "24",
      "Name": "NVMe Backplane",
      "Status": {
        "Health": "OK",
        "State": "Enabled"
      }
    },
...
}
```

### CODE REVIEW

File: redfish-core/lib/pcie_slots.hpp [hunk 1]
```diff
@@ -121,6 +121,89 @@ inline void
...
+            }
+        };
+
+        redfish::chassis_utils::getChassisAssembly(
+            asyncResp, chassisId, std::move(backplaneAssemblyCallback));
+    };
+
+    crow::connections::systemBus->async_method_call(
+        respHandler, "xyz.openbmc_project.ObjectMapper",
+        pcieSlotPath + "/inventory", "org.freedesktop.DBus.Properties", "Get",
```
Comment (gtmills) [inline redfish-core/lib/pcie_slots.hpp:203]: Same comments as https://github.com/ibm-openbmc/bmcweb/pull/576 
Up to you if you want to move towards the upstream style

File: redfish-core/lib/pcie_slots.hpp [hunk 2]
```diff
@@ -121,6 +121,89 @@ inline void
...
 
+/**
+ * @brief Add PCIeSlot to NMVe backplane assembly link
+ *
+ * @param[in, out]  asyncResp       Async HTTP response.
+ * @param[in]       pcieSlotPath    Object path of the PCIeSlot.
+ * @param[in]       index           Index.
+ */
+inline void linkAssociatedDiskBackplane(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
```
Comment (gtmills) [inline redfish-core/lib/pcie_slots.hpp:132]: include what you use, string, memory, vector, variant, etc

File: redfish-core/lib/pcie_slots.hpp (hunk 1 above)
Comment (baemyung) [reply redfish-core/lib/pcie_slots.hpp]: I changed a few places... 
I'll keep working on it.

Comment (baemyung) [reply redfish-core/lib/pcie_slots.hpp]: I've changed all of the occurrences of async_method_call() in pcie_slots.hpp

**Review by gtmills — APPROVED** [review]

**Comment by baemyung** [comment]
Additional fix is done via https://github.com/ibm-openbmc/bmcweb/pull/603

**Pull request merged**